libminifi/src/core/ProcessGroup.cpp

```diff
@@ -136,13 +147,14 @@ void ProcessGroup::removeProcessGroup(ProcessGroup *child) {
   }
 }
 
-void ProcessGroup::startProcessing(const std::shared_ptr<TimerDrivenSchedulingAgent> timeScheduler, const std::shared_ptr<EventDrivenSchedulingAgent> &eventScheduler,
-                                   const std::shared_ptr<CronDrivenSchedulingAgent> &cronScheduler) {
+void ProcessGroup::startProcessingProcessors(const std::shared_ptr<TimerDrivenSchedulingAgent> timeScheduler,
+    const std::shared_ptr<EventDrivenSchedulingAgent> &eventScheduler, const std::shared_ptr<CronDrivenSchedulingAgent> &cronScheduler) {
   std::lock_guard<std::recursive_mutex> lock(mutex_);
 
-  try {
-    // Start all the processor node, input and output ports
-    for (const auto &processor : processors_) {
+  std::set<std::shared_ptr<Processor> > failed_processors;
+
+  for (const auto &processor : failed_processors_) {
+    try {
```

Comment:
If I understand correctly, this means that processors that can be successfully scheduled will remain scheduled even if other processors fails to schedule.
This will cause processors before the failed ones to fill the connections until backpressure stops them, and processors after the failed ones to starve.
I am not saying this is a bad thing, I am asking if this is the behaviour we want/NiFi follows.

Reply:
In my opinion, yes, because of: 
-We can assume that in real production systems onSchedule errors are mostly transient (network, etc), so a retry is going to fix them. 
-What could we do to avoid such scenarios? We could do a BFS initiated from the failed processor and unschedule all the sources, but in my opinion that's even worse: it's quite complex and we just achieve immediately loosing data instead of buffering it to our connections.